dm/pkg/conn/baseconn.go

```diff
@@ -223,6 +225,34 @@ func (conn *BaseConn) ExecuteSQL(tctx *tcontext.Context, hVec *prometheus.Histog
 	return conn.ExecuteSQLWithIgnoreError(tctx, hVec, task, nil, queries, args...)
 }
 
+// ExecuteSQLsAutoSplit executes sqls and when meet "transaction too large" error,
+// it will try to split the sqls into two parts and execute them again.
+// The `queries` and `args` should be the same length.
+func (conn *BaseConn) ExecuteSQLsAutoSplit(
+	tctx *tcontext.Context,
+	hVec *prometheus.HistogramVec,
+	task string,
+	queries []string,
+	args ...[]interface{},
+) error {
+	_, err := conn.ExecuteSQL(tctx, hVec, task, queries, args...)
+	mysqlErr, ok := errors.Cause(err).(*mysql.MySQLError)
+	if !ok {
+		return err
+	}
+
+	if mysqlErr.Number != errno.ErrTxnTooLarge || len(queries) == 1 {
+		return err
+	}
+
+	mid := len(queries) / 2
+	err = conn.ExecuteSQLsAutoSplit(tctx, hVec, task, queries[:mid], args[:mid]...)
```

Comment:
this pr is part 1? seems no one calls `ExecuteSQLsAutoSplit`.

if checkpoint is half saved, will it cause `global pos`(save success) > `some table pos` which failed to save

Reply:
yes, I'll let syncer checkpoint uses `ExecuteSQLsAutoSplit` in another PR.

And DML of global checkpoint should be the last DML, so if checkpoint is half saved only table checkpoint is safely forwarded.